elements/pf-progress/pf-progress.ts

```diff
@@ -0,0 +1,228 @@
+ *
+ * @cssprop {<length>} --pf-c-progress--m-lg__bar--Height
+ *          Height of the progress bar when the size is large.
+ *          {@default `1.5rem`}
+ *
+ */
+@customElement('pf-progress')
+export class PfProgress extends LitElement {
+  static readonly styles = [styles];
+
+  static readonly formAssociated = true;
+
+  #calculatedPercentage = 0;
+
+  #logger = new Logger(this);
+
+  @property({ reflect: true, type: Number })
+    value = 0;
+
+  @property()
+    title = '';
+
+  @property({ attribute: 'aria-label' })
+    ariaLabel = '';
+
+  @property({ type: Number })
+    max = 100;
+
+  @property({ type: Number })
+    min = 0;
+
+  @property()
+    size: 'sm' | 'lg' | '' = '';
+
+  @property({ reflect: true, attribute: 'measure-location' })
+    measureLocation: '' | 'outside' | 'inside' | 'none' = '';
+
+  @property()
+    variant: '' | 'success' | 'danger' | 'warning' = '';
+
+  @query('progress')
+    progressElement!: HTMLProgressElement;
+
+  #showInsideStatus() {
+    return this.measureLocation === 'inside';
+  }
+
+  #calculatePercentage() {
+    const { value, max, min } = this;
+    if (value === null ) {
+      this.#logger.warn(`A progress element requires a value attribute.`);
+      return;
+    }
+    const percentage = (value - min) / (max - min);
+    this.#calculatedPercentage = Math.round(percentage * 100);
+  }
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    this.#calculatePercentage();
+    this.#updateAccessibility();
+  }
+
+  #updateAccessibility() {
+    const { value, title, ariaLabel, max, progressElement } = this;
+    if (progressElement) {
+      progressElement.setAttribute('aria-valuemin', '0');
+      if (title || ariaLabel) {
+        progressElement.setAttribute('aria-label', `${title ? title : ariaLabel}`);
+      }
+      if (max != null) {
+        progressElement.setAttribute('aria-valuemax', `${max}`);
+      }
+      if (value) {
+        progressElement.setAttribute('aria-valuenow', `${value}`);
+      }
+    }
+  }
+
+  render() {
+    const { size, measureLocation, variant, title } = this;
+    const icon = variant && ICONS.get(variant)?.icon;
+    const singleLine = title.length === 0 ? 'singleline' : '';
+
+    return html`
+      <div
+        class="container ${classMap({ [size]: !!size, [measureLocation]: !!measureLocation, [variant]: !!variant, [singleLine]: !!singleLine })}">
+        
+        ${title ?
+          html`
+          <div id="title" class="title">${title}</div>` : ''}
+          
+          ${measureLocation !== 'none' ? html`<div class="status">
+            ${html`${!this.#showInsideStatus() ? `${this.#calculatedPercentage}%` : ''}`}
+            <pf-icon set="fas"
+                  size="md"
+                  ?hidden="${!icon}"
+                  icon="${ifDefined(icon)}"></pf-icon>
+          </div>` : ''}
+            
+          <progress tabindex="0" max="100" value="${this.#calculatedPercentage}"></progress>
```

Comment:
@nikkimk ptal at this tabindex

Reply:
Removed the tabindex and the form association after chatting with @hellogreg, we will add it as something we can add in later but outside of https://www.magentaa11y.com/checklist-web/progress/ it does not appear that others recommend treating progress like an input element for a screen reader.